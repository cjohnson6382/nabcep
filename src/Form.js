import React from 'react'
import PropTypes from "prop-types"

const fieldTypes = {
	radio: change => r => (
		<div>
			{ r.options.map((c, i) => <div key={ i } style={ { display: "flex", flexDirection: "row", alignItems: "center" } } >
				<input type="radio" name={ r.name } value={ c.value } onChange={ change } />
				<div style={ { padding: "0 0 0 1em" } } >{ c.value }</div>
			</div> ) }
		</div>
	),
	text: change => () => <input type="text" onChange={ change } />,
	select: change => s => (
		<select onChange={ change } >
			{ s.options.map((c, i) => <option key={ i } value={ c.value } >{ c.name }</option>) }
		</select>
	),
	color: change => () => <input type="color" onChange={ change } />,
	date: change => () => <input type="date" onChange={ change } />,
	email: change => () => <input type="email" onChange={ change } />,
	password: change => () => <input type="password" onChange={ change } />,
	number: change => () => <input type="number" onChange={ change } />
}

const Field = ({ title, type, data, change }) => {
	let c = change(title)
	return (
		<div style={ { display: "flex", flexDirection: "row", padding: "1em 1em 1em 1em", alignItems: "center" } } >
			<label style={ { padding: "0 0.5em 0 0", alignItems: "center", display: "flex" } } >{ title }</label>
			{ fieldTypes[type](c)(data) }
		</div>
	)
}

const Form = ({ auth, submit, submitName, fields, title, bodyStyle, titleStyle, change }) => (
	<div style={ bodyStyle } >
		<h3 style={ titleStyle } >{ title }</h3>
		<form onSubmit={ submit } style={ { padding: "1em" } } >
			{ fields.map((f, i) => <Field key={ i } { ...{ ...f, change } } />) }
			<input style={ { background: "linear-gradient(#3979d2, #154c94)", color: "white", borders: "none", padding: "0.3em" } } type="submit" value={ submitName || "Submit" } />
		</form>
	</div>
)

Form.propTypes = {
	auth: PropTypes.object,
	submit: PropTypes.func.isRequired,
	submitName: PropTypes.string,
	fields: PropTypes.array.isRequired,
	title: PropTypes.string,
	titleStyle: PropTypes.object,
	bodyStyle: PropTypes.object
}

export default Form