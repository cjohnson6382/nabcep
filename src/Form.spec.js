import React from 'react'
import TestRenderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Form from './Form'

// jest.unmock('./utilities')
// import * as utilities from './utilities'

const propz = {
	auth: {},
	submit: jest.fn(),
	submitName: "submit",
	fields: [
		{ title: "field1", type: "text" },
		{ title: "field2", type: "radio", data: { options: [ { value: "first" }, { value: "second" } ] } },
		{ title: "field3", type: "select", data: { options: [ { name: "option1", value: "first" }, { name: "option2", value: "second" } ] } }
	],
	title: "test form",
	titleStyle: {
		padding: "1em",
		background: "linear-gradient(#b4b4b4, #cacaca, #b4b4b4)"
	},
	bodyStyle: {
		background: "#e5e5e5"
	},
	change: jest.fn()
}


describe('Form', () => {
	let f = null

	it('renders', () => {
		f = TestRenderer.create(<Form { ...propz } />)
		
		const tree = f.toJSON()
		expect(tree).toMatchSnapshot()
	})
	
	// it('renders correctly', () => {
	// 	expect(t.state().data).toEqual(r)
	// })

})