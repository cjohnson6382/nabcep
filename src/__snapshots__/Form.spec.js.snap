// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders 1`] = `
<div
  style={
    Object {
      "background": "#e5e5e5",
    }
  }
>
  <h3
    style={
      Object {
        "background": "linear-gradient(#b4b4b4, #cacaca, #b4b4b4)",
        "padding": "1em",
      }
    }
  >
    test form
  </h3>
  <form
    onSubmit={[Function]}
    style={
      Object {
        "padding": "1em",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "padding": "1em 1em 1em 1em",
        }
      }
    >
      <label
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "padding": "0 0.5em 0 0",
          }
        }
      >
        field1
      </label>
      <input
        onChange={undefined}
        type="text"
      />
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "padding": "1em 1em 1em 1em",
        }
      }
    >
      <label
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "padding": "0 0.5em 0 0",
          }
        }
      >
        field2
      </label>
      <div>
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <input
            name={undefined}
            onChange={undefined}
            type="radio"
            value="first"
          />
          <div
            style={
              Object {
                "padding": "0 0 0 1em",
              }
            }
          >
            first
          </div>
        </div>
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <input
            name={undefined}
            onChange={undefined}
            type="radio"
            value="second"
          />
          <div
            style={
              Object {
                "padding": "0 0 0 1em",
              }
            }
          >
            second
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "padding": "1em 1em 1em 1em",
        }
      }
    >
      <label
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "padding": "0 0.5em 0 0",
          }
        }
      >
        field3
      </label>
      <select
        onChange={undefined}
      >
        <option
          value="first"
        >
          option1
        </option>
        <option
          value="second"
        >
          option2
        </option>
      </select>
    </div>
    <input
      style={
        Object {
          "background": "linear-gradient(#3979d2, #154c94)",
          "borders": "none",
          "color": "white",
          "padding": "0.3em",
        }
      }
      type="submit"
      value="submit"
    />
  </form>
</div>
`;
